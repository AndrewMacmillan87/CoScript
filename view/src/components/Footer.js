import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";

// Redux
import { connect } from "react-redux";
import { getMostRecentProject } from "../redux/actions/documents";

import PropTypes from "prop-types";

import "./footer.css";

export const Footer = ({ user, loading, getMostRecentProject, ...props }) => {
    const authLinks = (
        <div className='footer-links'>
            <Link to='/'>Home</Link>
            <Link to='/profile'>Profile</Link>
            <Link
                to='#'
                onClick={async () => {
                    await getMostRecentProject();
                    const { projectData } = props;
                    projectData
                        ? props.history.push(`/editor${projectData.url}`)
                        : props.history.push("/editor");
                }}>
                Workspace
            </Link>
            <Link to='/projects'>Active Projects</Link>
            <Link to='/publications'>Publications</Link>
        </div>
    );

    const guestLinks = (
        <div className='footer-links'>
            <Link to='/'>Home</Link>
            <Link to='/projects'>Active Projects</Link>
            <Link to='/publications'>Publications</Link>
        </div>
    );

    const isLoggedIn = user ? user.isLoggedIn : false;

    return (
        <footer className='footer'>
            <h2 className='footer-h2'>CoScript</h2>
            <div className='footer-container'>
                <div className='links-container socials'>
                    <p>Socials</p>
                    <div className='footer-links'>
                        <a href='https://github.com/AndrewMacmillan87/coscript'>
                            GitHub
                        </a>
                        <a href='/'>Twitter</a>
                        <a href='/'>Facebook</a>
                    </div>
                </div>
                <div className='links-container site-map'>
                    <p>Site Map</p>

                    {!loading && (
                        <Fragment>
                            {isLoggedIn ? authLinks : guestLinks}
                        </Fragment>
                    )}
                </div>
                <div className='links-container legals'>
                    <p>Legal</p>
                    <div className='footer-links'>
                        <a href='/terms'>Terms and Conditions</a>
                        <a href='/copyright'>Copyright Info</a>
                        <a href='/agreement'>Writers Agreements</a>
                        <a href='/'>Accreditation</a>
                    </div>
                </div>
            </div>
        </footer>
    );
};

Footer.propTypes = {
    user: PropTypes.object,
    loading: PropTypes.bool,
    getMostRecentProject: PropTypes.func.isRequired,
};

const mapProps = (state) => ({
    user: state.auth.user,
    loading: state.auth.loading,
    projectData: state.documents.projectData,
});

export default connect(mapProps, { getMostRecentProject })(withRouter(Footer));
