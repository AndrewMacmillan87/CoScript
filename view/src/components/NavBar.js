import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";
import Button from "./buttons/Button";

import { getUserMessages } from "../redux/auxiliary/user-messages";

// Redux
import { connect } from "react-redux";
import { getMostRecentProject } from "../redux/actions/documents";

import PropTypes from "prop-types";

import "./navbar.css";

export class NavBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            visible: false,
            unread: false,
        };
        this.handleClick = this.handleClick.bind(this);
    }

    /**
     * Checks to see if the user has any unread messages
     * in their personal profile message inbox and sets
     * the 'unread' state property to true if they have.
     * This displays a small icon to the user to indicate
     * they have unread messages
     */
    async componentDidMount() {
        const userMessages = await getUserMessages();
        for (let message of userMessages) {
            if (!message.read) {
                this.setState({
                    unread: true,
                });
            }
        }
    }

    /**
     * Sets the 'visible' state property so that the
     * dropdown menu can be revealed and hidden by
     * clicking the link
     */
    handleClick() {
        this.setState((state) => ({
            visible: !state.visible,
        }));
    }

    render() {
        const { unread } = this.state;
        const { user, loading, history } = this.props;

        const menuIcon = require("../img/menu.svg");
        const spinner = require("../img/loading.svg");
        const isLoggedIn = user && user.isLoggedIn | false;

        const loggedIn = (
            <Fragment>
                <div className='header'>
                    <div className='logo-container'>
                        <Link to='/'>
                            <p className='logo'>CoScript</p>
                        </Link>
                    </div>
                    <div className='nav-links'>
                        <Link to='/projects'>Active Projects</Link>
                        <Link to='/publications'>Publications</Link>
                        <Link
                            to='#'
                            onClick={async () => {
                                await this.props.getMostRecentProject();
                                const { projectData } = this.props;
                                projectData
                                    ? history.push(`/editor${projectData.url}`)
                                    : history.push("/editor");
                            }}>
                            Workspace
                        </Link>
                    </div>
                </div>
                <div className='nav-btns'>
                    {user && !loading ? (
                        <div className='logged-in-user-container'>
                            <p className='user-username'>
                                {user.profile.displayName}
                            </p>
                            <button
                                className='desktop-icon-link'
                                onClick={this.handleClick}>
                                {unread ? (
                                    <div className='avatar-unread-msg-container'>
                                        <div className='avatar-unread-msg-icon'></div>
                                        <img
                                            className='user-gravatar'
                                            src={user.avatar}
                                            alt='user gravatar'
                                        />
                                    </div>
                                ) : (
                                    <img
                                        className='user-gravatar'
                                        src={user.avatar}
                                        alt='user gravatar'
                                    />
                                )}
                            </button>
                        </div>
                    ) : (
                        <img
                            className='user-spinner'
                            src={spinner}
                            alt='loading spinner icon'
                        />
                    )}
                </div>
                <button
                    className='dropdown-icon-link'
                    onClick={this.handleClick}>
                    <img
                        className='dropdown-icon'
                        src={menuIcon}
                        alt='Burger menu icon'
                    />
                </button>
                <div
                    id='dropdown-nav'
                    className={
                        this.state.visible
                            ? "dropdown-menu dropdown-content display"
                            : "dropdown-menu dropdown-content"
                    }>
                    <Link to='/projects'>
                        <p className='dropdown-link'>Active Projects</p>
                    </Link>
                    <Link to='/publications'>
                        <p className='dropdown-link'>Publications</p>
                    </Link>
                    <Link
                        to='#'
                        onClick={async () => {
                            await this.props.getMostRecentProject();
                            const { projectData } = this.props;
                            projectData
                                ? history.push(`/editor${projectData.url}`)
                                : history.push("/editor");
                        }}>
                        <p className='dropdown-link'>Workspace</p>
                    </Link>
                    <Link to='/profile'>
                        <p className='dropdown-link'>My Profile</p>
                    </Link>
                    <a
                        onClick={() => {
                            history["from"] = "logout";
                            history.push("/redirect");
                        }}
                        href=' '>
                        <p className='dropdown-link'>Logout</p>
                    </a>
                </div>
            </Fragment>
        );

        const visiting = (
            <Fragment>
                <div className='header'>
                    <div className='logo-container'>
                        <Link to='/'>
                            <p className='logo'>CoScript</p>
                        </Link>
                    </div>
                    <div className='nav-links'>
                        <Link to='/projects'>Active Projects</Link>
                        <Link to='/publications'>Publications</Link>
                    </div>
                </div>
                <div className='nav-btns'>
                    <div>
                        <Link to='/register'>
                            <Button classes='btn' value='Register' />
                        </Link>
                    </div>
                    <div>
                        <Link to='/login'>
                            <Button classes='grn-btn' value='Login' />
                        </Link>
                    </div>
                </div>
                <button
                    className='dropdown-icon-link'
                    onClick={this.handleClick}>
                    <img
                        className='dropdown-icon'
                        src={menuIcon}
                        alt='Burger menu icon'
                    />
                </button>
                <div
                    id='dropdown-nav'
                    className={
                        this.state.visible
                            ? "dropdown-menu dropdown-content display"
                            : "dropdown-menu dropdown-content"
                    }>
                    <Link to='/register'>
                        <p className='dropdown-link'>Register</p>
                    </Link>
                    <Link to='/login'>
                        <p className='dropdown-link'>Login</p>
                    </Link>
                    <Link to='/projects'>
                        <p className='dropdown-link'>Active Projects</p>
                    </Link>
                    <Link to='/publications'>
                        <p className='dropdown-link'>Publications</p>
                    </Link>
                </div>
            </Fragment>
        );

        return (
            <nav className='navbar'>
                {!loading && (
                    <Fragment>{!isLoggedIn ? visiting : loggedIn}</Fragment>
                )}
            </nav>
        );
    }
}

NavBar.propTypes = {
    user: PropTypes.object,
    loading: PropTypes.bool,
    getMostRecentProject: PropTypes.func.isRequired,
};

const setProps = (state) => ({
    user: state.auth.user,
    loading: state.auth.loading,
    projectData: state.documents.projectData,
});

export default connect(setProps, {
    getMostRecentProject,
})(withRouter(NavBar));
