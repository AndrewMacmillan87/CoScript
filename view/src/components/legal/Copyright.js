import React, { Fragment, useEffect } from "react";

import NavBar from "../NavBar";
import Footer from "../Footer";

import "./legal.css";

export default function Copyright() {
    useEffect(() => {
        document.title = "CoScript - Copyright";
        window.scrollTo(0, 0);
    }, []);

    const wikiHowImage = require("../../img/copyright.PNG");
    const pwImage = require("../../img/copyright2.PNG");
    const wikipediaImage = require("../../img/copyright3.PNG");

    return (
        <Fragment>
            <NavBar />
            <main className='copyright-main'>
                <header className='copyright-header'>
                    <h1 className='copyright-h1'>
                        Copyright &copy; What you Should Know
                    </h1>
                </header>
                <p className='copyright-introduction'>
                    The work you make is valuable to you. You, and the people
                    you work with, spent a long time putting it together and
                    coming up with the ideas that layed the foundations for your
                    creation. Getting your work Copyrighted is not as
                    complicated as some might think and provides you with the
                    peace of mind that the work you create is attributed to you
                    and your coworkers. The following sources provide
                    information on what a Copyright is and how you can ensure
                    your work is protected. This is not obligatory but it is
                    recommended, so please have a read through these sources.
                </p>
                <p className='copyright-introduction-note'>
                    If you think these sources could be expanded, please email
                    suggestions to{" "}
                    <a href='mailto:legal@cosript.co.uk'>legal@cosript.co.uk</a>
                </p>
                <div className='source-info-container'>
                    <h2 className='source-info-header'>Wiki-How</h2>
                    <a
                        href='https://www.wikihow.com/Copyright-Your-Writing-for-Free'
                        rel='noopener noreferrer'
                        target='_blank'>
                        <img
                            className='source-link-img'
                            src={wikiHowImage}
                            alt='wiki-how website landing page'
                        />
                    </a>
                    <p className='source-info-description'>
                        A simple and well laid out, step-by-step guide to
                        understanding and putting in place a Copyright{" "}
                        <a
                            href='https://www.wikihow.com/Copyright-Your-Writing-for-Free'
                            rel='noopener noreferrer'
                            target='_blank'>
                            https://www.wikihow.com/Copyright-Your-Writing-for-Free
                        </a>
                    </p>
                </div>
                <div className='source-info-container'>
                    <h2 className='source-info-header'>Wikipedia</h2>
                    <a
                        href='https://en.wikipedia.org/wiki/Copyright'
                        rel='noopener noreferrer'
                        target='_blank'>
                        <img
                            className='source-link-img'
                            src={wikipediaImage}
                            alt='wiki-how website landing page'
                        />
                    </a>
                    <p className='source-info-description'>
                        In depth description of the history and purpose of
                        Copyright, as well as coverage of legal enforcement,
                        consequences of breaking Copyright and its limitations{" "}
                        <a
                            href='https://en.wikipedia.org/wiki/Copyright'
                            rel='noopener noreferrer'
                            target='_blank'>
                            https://en.wikipedia.org/wiki/Copyright
                        </a>
                    </p>
                </div>
                <div className='source-info-container'>
                    <h2 className='source-info-header'>pw.org</h2>
                    <a
                        href='https://www.pw.org/content/copyright'
                        rel='noopener noreferrer'
                        target='_blank'>
                        <img
                            className='source-link-img'
                            src={pwImage}
                            alt='pw.org website landing page'
                        />
                    </a>
                    <p className='source-info-description'>
                        A coverage of how Copyright is implemented in the U.S.
                        This may not be relevant to everyone reading this but
                        the site contains many useful links and articles that
                        could be helpful to new writers{" "}
                        <a
                            href='https://www.pw.org/content/copyright'
                            rel='noopener noreferrer'
                            target='_blank'>
                            https://www.pw.org/content/copyright
                        </a>
                    </p>
                </div>
            </main>
            <Footer />
        </Fragment>
    );
}
