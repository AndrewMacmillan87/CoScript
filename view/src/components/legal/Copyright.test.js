import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import Copyright from "./Copyright";
import NavBar from "../NavBar";
import Footer from "../Footer";

describe("Copyright component", () => {
    test("renders without crashing", () => {
        global.HTMLMediaElement.prototype.pause = jest.fn();
        global.scrollTo = jest.fn();
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        global.HTMLMediaElement.prototype.pause = jest.fn();
        global.scrollTo = jest.fn();
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <Copyright />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render the NavBar component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>
        );
        expect(wrapper.find(NavBar).length).toEqual(1);
    });

    it("should render a main element with a className of copyright-main", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>
        );
        expect(wrapper.find("main").length).toEqual(1);
        expect(wrapper.find({ className: "copyright-main" }).length).toEqual(1);
    });

    it("should render a header element with a className of copyright-header", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>
        );
        expect(wrapper.find("header").length).toEqual(1);
        expect(wrapper.find({ className: "copyright-header" }).length).toEqual(
            1
        );
    });

    it("should render 8 p elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>
        );
        expect(wrapper.find("p").length).toEqual(8);
    });

    it("should render 4 h2 elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>
        );
        expect(wrapper.find("h2").length).toEqual(4);
    });

    it("should render 14 a elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>
        );
        expect(wrapper.find("a").length).toEqual(14);
    });

    it("should render 3 img elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>
        );
        expect(wrapper.find("img").length).toEqual(3);
    });

    it("should render the Footer component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Copyright />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Footer).length).toEqual(1);
    });
});
