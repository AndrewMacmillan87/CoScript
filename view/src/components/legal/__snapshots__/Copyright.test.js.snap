// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright component renders correctly 1`] = `
Array [
  <nav
    className="navbar"
  />,
  <main
    className="copyright-main"
  >
    <header
      className="copyright-header"
    >
      <h1
        className="copyright-h1"
      >
        Copyright Â© What you Should Know
      </h1>
    </header>
    <p
      className="copyright-introduction"
    >
      The work you make is valuable to you. You, and the people you work with, spent a long time putting it together and coming up with the ideas that layed the foundations for your creation. Getting your work Copyrighted is not as complicated as some might think and provides you with the peace of mind that the work you create is attributed to you and your coworkers. The following sources provide information on what a Copyright is and how you can ensure your work is protected. This is not obligatory but it is recommended, so please have a read through these sources.
    </p>
    <p
      className="copyright-introduction-note"
    >
      If you think these sources could be expanded, please email suggestions to
       
      <a
        href="mailto:legal@cosript.co.uk"
      >
        legal@cosript.co.uk
      </a>
    </p>
    <div
      className="source-info-container"
    >
      <h2
        className="source-info-header"
      >
        Wiki-How
      </h2>
      <a
        href="https://www.wikihow.com/Copyright-Your-Writing-for-Free"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="wiki-how website landing page"
          className="source-link-img"
          src="copyright.PNG"
        />
      </a>
      <p
        className="source-info-description"
      >
        A simple and well laid out, step-by-step guide to understanding and putting in place a Copyright
         
        <a
          href="https://www.wikihow.com/Copyright-Your-Writing-for-Free"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://www.wikihow.com/Copyright-Your-Writing-for-Free
        </a>
      </p>
    </div>
    <div
      className="source-info-container"
    >
      <h2
        className="source-info-header"
      >
        Wikipedia
      </h2>
      <a
        href="https://en.wikipedia.org/wiki/Copyright"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="wiki-how website landing page"
          className="source-link-img"
          src="copyright3.PNG"
        />
      </a>
      <p
        className="source-info-description"
      >
        In depth description of the history and purpose of Copyright, as well as coverage of legal enforcement, consequences of breaking Copyright and its limitations
         
        <a
          href="https://en.wikipedia.org/wiki/Copyright"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://en.wikipedia.org/wiki/Copyright
        </a>
      </p>
    </div>
    <div
      className="source-info-container"
    >
      <h2
        className="source-info-header"
      >
        pw.org
      </h2>
      <a
        href="https://www.pw.org/content/copyright"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="pw.org website landing page"
          className="source-link-img"
          src="copyright2.PNG"
        />
      </a>
      <p
        className="source-info-description"
      >
        A coverage of how Copyright is implemented in the U.S. This may not be relevant to everyone reading this but the site contains many useful links and articles that could be helpful to new writers
         
        <a
          href="https://www.pw.org/content/copyright"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://www.pw.org/content/copyright
        </a>
      </p>
    </div>
  </main>,
  <footer
    className="footer"
  >
    <h2
      className="footer-h2"
    >
      CoScript
    </h2>
    <div
      className="footer-container"
    >
      <div
        className="links-container socials"
      >
        <p>
          Socials
        </p>
        <div
          className="footer-links"
        >
          <a
            href="https://github.com/AndrewMacmillan87/coscript"
          >
            GitHub
          </a>
          <a
            href="/"
          >
            Twitter
          </a>
          <a
            href="/"
          >
            Facebook
          </a>
        </div>
      </div>
      <div
        className="links-container site-map"
      >
        <p>
          Site Map
        </p>
      </div>
      <div
        className="links-container legals"
      >
        <p>
          Legal
        </p>
        <div
          className="footer-links"
        >
          <a
            href="/terms"
          >
            Terms and Conditions
          </a>
          <a
            href="/copyright"
          >
            Copyright Info
          </a>
          <a
            href="/agreement"
          >
            Writers Agreements
          </a>
          <a
            href="/"
          >
            Accreditation
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;
